app/controllers/publish_controller.rb

```diff
@@ -4,9 +4,12 @@ class PublishController < FormController
   def index
     @published_dev = PublishServicePresenter.new(publishes_dev, service)
     @published_production = PublishServicePresenter.new(publishes_production, service)
+    declarations
   end
 
   def create
+    return unless can_publish_to_live || publish_service_params[:deployment_environment] == 'dev'
```

Comment:
Added this just to prevent hitting the route with a payload to bypass the publishing restriction.